app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixDateTimePicker.kt

```diff
@@ -0,0 +1,226 @@
+package com.arygm.quickfix.ui.elements
+
+import android.app.TimePickerDialog
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.AccessTime
+import androidx.compose.material.icons.filled.Keyboard
+import androidx.compose.material3.*
+import androidx.compose.material3.MaterialTheme.colorScheme
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.arygm.quickfix.utils.MyAppTheme
+import com.maxkeppeker.sheets.core.models.base.Header
+import com.maxkeppeker.sheets.core.models.base.rememberUseCaseState
+import com.maxkeppeler.sheets.calendar.CalendarDialog
+import com.maxkeppeler.sheets.calendar.models.CalendarConfig
+import com.maxkeppeler.sheets.calendar.models.CalendarSelection
+import com.maxkeppeler.sheets.calendar.models.CalendarStyle
+import java.time.LocalDate
+import java.time.LocalTime
+import java.util.Locale
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun QuickFixDateTimePicker(
+    onDateTimeSelected: (LocalDate, LocalTime) -> Unit,
+    onDismissRequest: () -> Unit,
+    modifier: Modifier = Modifier,
+    disableDates: List<LocalDate> = emptyList(),
+    timeViewTest: Boolean = true
+) {
+  var selectedDate by remember { mutableStateOf<LocalDate?>(null) }
+  var selectedTime by remember { mutableStateOf<LocalTime?>(null) }
+  var showDatePicker by remember { mutableStateOf(true) }
+  var showTimePicker by remember { mutableStateOf(false) }
+
+  if (showDatePicker) {
+    DatePickerDialog(
+        onDismissRequest = {
+          // Cancel the flow if dismissed
+          showDatePicker = false
+          onDismissRequest()
+        },
+        onDateSelected = { year, month, day -> selectedDate = LocalDate.of(year, month, day) },
+        onFinishedRequest = {
+          // Transition only if a date is selected
+          showDatePicker = false
+          showTimePicker = true
+        },
+        disableDates = disableDates)
+  }
+
+  if (showTimePicker) {
+    TimePickerDialog(
+        onDismissRequest = {
+          showTimePicker = false
+          onDismissRequest()
+        },
+        onTimeSelected = { hour, minute ->
+          selectedTime = LocalTime.of(hour, minute)
+          onDateTimeSelected(selectedDate!!, selectedTime!!)
```

Comment:
I think you should check that a date has indeed been picked, otherwise this could run the onDateTimeSelected with null dates.

Reply:
It is checked on the `TimePickerDialog` just below it with the `rememberTimePickerState` 😁